staff/staff_views_helper.py

```diff
@@ -208,3 +213,150 @@ def add_volunteers_to_volunteer_dict(classroom):
+    class_offering = get_class_offering_by_id(request.POST["course_id"])
+    classroom = Classroom.objects.get(id=request.POST["course_id"])
+    classroom.volunteers.remove(request.POST["former_vol"])
+    remove_enrollment(vol_contact, class_offering)
+
+
+def remove_student(request):
+    student_contact = get_contact_by_user_id(request.POST["former_student"])
+    class_offering = get_class_offering_by_id(request.POST["course_id"])
+    classroom = Classroom.objects.get(id=request.POST["course_id"])
+    classroom.students.remove(request.POST["former_student"])
+    remove_enrollment(student_contact, class_offering)
+
+
+def add_volunteers(request):
+    form = AddVolunteersForm(request.POST)
+    class_offering = get_class_offering_by_id(request.POST["course_id"])
+    classroom = Classroom.objects.get(id=request.POST["course_id"])
+    if form.is_valid():
+        for volunteer in form.cleaned_data.get("volunteers"):
+            vol_contact = get_contact_by_user_id(volunteer)
+            ClassEnrollment.objects.create(
+                created_by=class_offering.created_by,
+                contact=vol_contact,
+                status="Enrolled",
+                class_offering=class_offering,
+            )
+            classroom.volunteers.add(volunteer)
+
+
+def add_students(request):
+    form = AddStudentsForm(request.POST)
+    class_offering = get_class_offering_by_id(request.POST["course_id"])
+    classroom = Classroom.objects.get(id=request.POST["course_id"])
+    if form.is_valid():
+        for student in form.cleaned_data.get("students"):
+            student_contact = get_contact_by_user_id(student)
+            ClassEnrollment.objects.create(
+                created_by=class_offering.created_by,
+                contact=student_contact,
+                status="Enrolled",
+                class_offering=class_offering,
+            )
+            classroom.students.add(student)
+
+
+def remove_enrollment(contact, class_offering):
+    old_enrollment = ClassEnrollment.objects.get(
+        contact=contact, status="Enrolled", class_offering=class_offering
+    )
+    created_by = old_enrollment.created_by
+    old_enrollment.delete()
+    return created_by
+
+
+def get_contact_by_user_id(id):
+    return Contact.objects.get(
+        client_id=UserProfile.objects.get(user_id=id).salesforce_id
+    )
+
+
+def get_class_offering_by_id(id):
+    course_name = get_course_name_by_id(id)
+    return ClassOffering.objects.get(name=course_name)
+
+
+def get_course_name_by_id(id):
+    return Classroom.objects.get(id=id).course
+
+
+def sync_attendance_with_salesforce_class_offerings():
+    """
+    This method is fluid; in its current state it will duplicate entries
+    in your postgres templates database if you've already got templates data!
+    Only use if you don't have templates data but do have Classroom
+    data.
+    """
+    for classoffering in ClassOffering.objects.all():
```

Comment:
What's the rationale for creating all of these in batch rather than on demand when attendance is taken?

Reply:
This method will not be used in production, I should've delete it, it was to update my existing postresdb that was already filled up with data I had been using to work with functionality.